src/hooks/useQuery/useQuery.ts

```diff
@@ -1,21 +1,23 @@
-import { QueryKey, UseQueryOptions, useQuery as useRQQuery } from '@tanstack/react-query';
+import { QueryFunction, QueryKey, UseQueryOptions, useQuery as useRQQuery } from '@tanstack/react-query';
+import { AxiosInstance } from 'axios';
 
 import { useApiClient } from '../useApiClient/useApiClient';
-import { AxiosQueriesType, queries } from 'api/actions';
-import { DataForQuery, ExtendedQueryMeta, GetQueryParams } from 'api/types/types';
-import { parseQueryKey } from 'utils/parseQueryKey';
-import { ApiError } from 'context/apiClient/apiClientContextController/apiError/apiError.types';
+import { ExtendedQueryMeta } from 'api/types/types';
+import { StandardizedApiError } from 'context/apiClient/apiClientContextController/apiError/apiError.types';
 
-export const useQuery = <Key extends keyof AxiosQueriesType, TError = ApiError>(
-  query: Key,
-  args: GetQueryParams<Key>,
-  options?: UseQueryOptions<DataForQuery<Key>, TError> & { meta?: Partial<ExtendedQueryMeta> },
+export const useQuery = <TQueryFnData = unknown, TError = StandardizedApiError>(
+  params: Omit<UseQueryOptions<TQueryFnData, TError>, 'queryFn'> & {
+    meta?: Partial<ExtendedQueryMeta>;
+    queryFn: (client: AxiosInstance) => QueryFunction<TQueryFnData, QueryKey>;
+  },
 ) => {
   const { client } = useApiClient();
-  const queryFn = queries[query](client);
-  const queryKey: QueryKey = parseQueryKey(query, args);
+  const { queryFn, ...options } = params;
 
-  const result = useRQQuery(queryKey, async () => (await queryFn(args)) as DataForQuery<Key>, options);
+  const result = useRQQuery({
+    queryFn: (args) => queryFn(client)(args),
+    ...options,
+  });
 
   return { ...result, isLoadingAndEnabled: result.isLoading && result.fetchStatus !== 'idle' };
```

Comment:
```suggestion
  return { ...result, isLoadingAndEnabled: result.isPending && result.fetchStatus !== 'idle' };
```
Shouldn't it be changed like that cause `isLoading` was renamed to `isPending` and now `isLoading` is the previous `isInitialLoading`